src/test/kotlin/uk/gov/dluhc/printapi/jobs/FinalRetentionPeriodDataRemovalJobIntegrationTest.kt

```diff
@@ -2,46 +2,70 @@ package uk.gov.dluhc.printapi.jobs
+        val certificate1 = buildCertificate(
+            sourceReference = "6407b6158f529a11713a1e5c",
+            finalRetentionRemovalDate = LocalDate.now().minusDays(1),
+            applicationReference = "arn:aws:s3:::$s3Bucket/$s3PathPhoto1", // TODO EIP1-4307 - switch to photoLocationArn once in place
+        )
+        val s3PathPhoto2 = anotherPhotoBucketPath()
+        val certificate2 = buildCertificate(
+            sourceReference = "2304v5134f529a11713a1e6a",
+            finalRetentionRemovalDate = LocalDate.now().minusDays(1),
+            applicationReference = "arn:aws:s3:::$s3Bucket/$s3PathPhoto2", // TODO EIP1-4307 - switch to photoLocationArn once in place
+        )
+        val certificate3 = buildCertificate(finalRetentionRemovalDate = LocalDate.now()) // should not be removed until tomorrow
+        val certificate4 = buildCertificate(finalRetentionRemovalDate = LocalDate.now().plusDays(1)) // should not be removed
         certificateRepository.saveAll(listOf(certificate1, certificate2, certificate3, certificate4))
+
+        s3Client.addCertificatePhotoToS3(s3Bucket, s3PathPhoto1)
+        s3Client.addCertificatePhotoToS3(s3Bucket, s3PathPhoto2)
         TestLogAppender.reset()
 
         // When
         finalRetentionPeriodDataRemovalJob.removeVoterCardFinalRetentionPeriodData()
 
         // Then
-        assertThat(certificateRepository.findById(certificate1.id!!)).isEmpty
-        assertThat(certificateRepository.findById(certificate2.id!!)).isEmpty
-        assertThat(certificateRepository.findById(certificate3.id!!)).isNotEmpty
-        assertThat(certificateRepository.findById(certificate4.id!!)).isNotEmpty
-        assertThat(testPrintRequestRepository.findById(certificate1.printRequests[0].id!!)).isEmpty
-        assertThat(testPrintRequestRepository.findById(certificate2.printRequests[0].id!!)).isEmpty
-        assertThat(testPrintRequestRepository.findById(certificate3.printRequests[0].id!!)).isNotEmpty
-        assertThat(testPrintRequestRepository.findById(certificate4.printRequests[0].id!!)).isNotEmpty
-        assertThat(TestLogAppender.hasLog("Removed remaining data after final retention period from certificate with sourceReference ${certificate1.sourceReference}", Level.INFO)).isTrue
-        assertThat(TestLogAppender.hasLog("Removed remaining data after final retention period from certificate with sourceReference ${certificate2.sourceReference}", Level.INFO)).isTrue
+        await.atMost(5, TimeUnit.SECONDS).untilAsserted {
+            assertThat(certificateRepository.findById(certificate1.id!!)).isEmpty
+            assertThat(certificateRepository.findById(certificate2.id!!)).isEmpty
+            assertThat(certificateRepository.findById(certificate3.id!!)).isNotEmpty
+            assertThat(certificateRepository.findById(certificate4.id!!)).isNotEmpty
+            assertThat(testPrintRequestRepository.findById(certificate1.printRequests[0].id!!)).isEmpty
```

Comment:
You could use `CertificateAssert` class if required. It has `hasNoPrintRequests` function.

Reply:
I'm making sure the print_request has been removed from the database entirely, rather than just asserting the `PrintRequest` has been removed from the `Certificate` entity. For this reason, I'm using `TestPrintRequestRepository` which I added in a previous PR (I didn't want to add a `PrintRequestRepository` in the main scope).